src/NugetFeedDeploymentRepository/NugetFeedApplicationRepository.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.Storage;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Etg.Yams.NuGet.Storage
+{
+    public class NugetFeedApplicationRepository : IApplicationRepository
+    {
+        private readonly SourceRepository _sourceRepository;
+
+        public NugetFeedApplicationRepository(string feedUrl = "https://api.nuget.org/v3/index.json", NugetFeedCredentials credentials = null)
+        {
+            this.FeedUrl = feedUrl;
+
+            List<Lazy<INuGetResourceProvider>> providers = new List<Lazy<INuGetResourceProvider>>();
+            providers.AddRange(Repository.Provider.GetCoreV3());
+            PackageSource packageSource = new PackageSource(feedUrl);
```

Comment:
Bonus: Ideally, this would be behind an interface and injected to this class (IoC) so we can use a mock to test edge cases such as what happens if the nuget repo throw a particular exception, the feed is down, etc. It's a bit involved so I'll leave it up to you.

Reply:
I will work on it and add tests.